src/config.rs

```diff
@@ -0,0 +1,100 @@
+//! Configuration for `HttpClient`s.
+
+use std::sync::Arc;
+use std::{collections::HashMap, fmt::Debug, time::Duration};
+
+use http_client::{Config as ClientConfig, HttpClient};
+use http_types::headers::{HeaderName, HeaderValues, ToHeaderValues};
+
+use crate::http::Url;
+use crate::Result;
+
+/// Configuration for `surf::Client`s and their underlying HTTP clients.
+#[non_exhaustive]
+#[derive(Clone, Debug)]
+pub struct Config {
+    /// The base URL for a client. All request URLs will be relative to this URL.
+    ///
+    /// Note: a trailing slash is significant.
+    /// Without it, the last path component is considered to be a “file” name
+    /// to be removed to get at the “directory” that is used as the base.
```

Comment:
Could we just add a slash at the end for them and remove this footgun? Is there ever a case where we wouldn't want that?

Reply:
Probably? This issue is tracked at https://github.com/http-rs/surf/issues/255